doc/releases/changelog-dev.md

```diff
@@ -8,11 +8,17 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The `__eq__` and `__hash__` methods of `Operator` and `MeasurementProcess` no longer rely on the
```

Comment:
Might be worth adding some examples for those unused to defining python dunder methods.  We could potentially also advertise it as an improvement.

Reply:
I added examples. In my opinion it should still be a breaking change, since we are changing behaviour dramatically.